packages/api/src/hooks/useResidenceList.ts

```diff
@@ -0,0 +1,24 @@
+import { useMemo } from 'react';
+import useQuery from '../useQuery';
+
+/** A custom hook that gets the residence list. */
+const useResidenceList = () => {
+    const { data: residence_list, ...residence_list_rest } = useQuery('residence_list');
+
+    /** Modify the residence list data. */
+    const modified_residence_list = useMemo(() => {
+        if (!residence_list || !residence_list.residence_list) return [];
+
+        return residence_list.residence_list.map(residence => ({
```

Comment:
`residence_list.residence_list` seems to be confusing. 
I'm not sure about the pattern you are following but I think it should be like `response.residence_list`

Reply:
done